# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1395978122 14400
#      Thu Mar 27 23:42:02 2014 -0400
# Branch stable
# Node ID 2633b5f3106a6b6ab8bedaca9a7070663322dc52
# Parent  3e53eca42f00651f7978f1d09fe088d9d01c3938
__makeinfo__: Process Texinfo docstrings without type conversion (bug #41965)

* __makeinfo__.m: Use fprintf instead of fwrite to pass Texinfo docstrings to
the makeinfo program without type conversion.

diff --git a/scripts/help/__makeinfo__.m b/scripts/help/__makeinfo__.m
--- a/scripts/help/__makeinfo__.m
+++ b/scripts/help/__makeinfo__.m
@@ -118,17 +118,17 @@ function [retval, status] = __makeinfo__
 
   unwind_protect
     ## Write Texinfo to tmp file
     template = "octave-help-XXXXXX";
     [fid, name] = mkstemp (fullfile (tempdir, template), true);
     if (fid < 0)
       error ("__makeinfo__: could not create temporary file");
     endif
-    fwrite (fid, text);
+    fprintf (fid, "%s", text);
     fclose (fid);
 
     ## Take action depending on output type
     switch (lower (output_type))
       case "plain text"
         cmd = sprintf ("%s --no-headers --no-warn --force --no-validate --output=- %s",
                        makeinfo_program (), name);
       case "html"
